### PR TITLE
Sockets - Erica

### PR DESCRIPTION
**Pull request by norrise120**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | When running into bugs (and there were a lot), I was unable to use something like 'pry' or the vs code debuggger like I would have normally have used in Ruby.  However, I was able to throw in 'console.log' statements similarly to how I would use 'puts' statements.  This turned out to be a relatively efficient way of debugging.
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | Yes.  In the beginning with ruby, I was able to answer a lot of my questions by just reading the ruby docs.  So far, in order to answer seemingly simple questions (like how to delete an element from an array), I had to check several different sites to get the answer for my particular problem.
What was a challenge you were able to overcome on this assignment? | One challenge that I ran into with this assignment was just accepting how verbose JS can be.  With my first method 'drawLetters', I felt that my code was very long and overly complicated.  I discussed it with Devin and we came to the conclusion that JS is missing a lot of the helpful functions that Ruby has, making the necessary code more complex than its Ruby counterpart. 
What has been interesting and positive about learning a new programming language? | Something positive about learning a new programming language is how much overlap there is! We were able to learn the basics in JS soooo much faster than we spent first learning basics in Ruby.
What is something to focus on your learnings in JavaScript in the next week? | I would like to learn more about different debugging methods.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,8 +1,98 @@
...
+    let drawnLetters = []
+    //Logic for drawing the letters
+    for (let i = 0; i < 10; i += 1) {
+      let letter = letters[Math.floor(Math.random() * (letters.length))];
+      drawnLetters.push(letter);
+
+      //This will remove the drawn letter just once from the letters array
+      for (let i = 0; i < letters.length; i += 1) {
+        if (letters[i] === letter) {
+          letters.splice(i, 1);
```
Comment (dHelmgren) [inline src/adagrams.js:28]: nice!

**Comment by dHelmgren** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | Good. Hope you like debugging in browser
Small commits with meaningful commit messages | yes, id like mORE tho.
**Code Requirements** |
`drawLetters` method | yes
Uses appropriate data structure to store the letter distribution | yes
All tests for `drawLetters` pass | yes
`usesAvailableLetters` method | yes
All tests for `usesAvailableLetters` pass | yes
`scoreWord` method | yes
Uses appropriate data structure to store the letter scores | yes
All tests for `scoreWord` pass | yes
`highestScoreFrom` method | yes
Appropriately handles edge cases for tie-breaking logic | yes
All tests for `highestScoreFrom` pass | yes
**Overall** | Great work on this project! In this project you’ve taken some interesting logic and worked it into JavaScript syntax. I’m adding a few comments on places that this could be better and minor suggestions for improvement, but overall, you’ve done a great job of practicing syntax.
